api/api.go

```diff
@@ -168,6 +168,9 @@ func makeHandler(ctrl libnetwork.NetworkController, fct processor) http.HandlerF
 		}
 
 		res, rsp := fct(ctrl, mux.Vars(req), body)
+		if !rsp.isOK() {
+			res = rsp.Status
```

Comment:
I see that this was removed in one of the recent changes inadvertently.

But I think we should replace this with 
`http.Error(w, rsp.Status, rsp.StatusCode)`
Can you please try it ?